src/main/java/hu/akarnokd/rxjava2/operators/FlowableTransformers.java

```diff
@@ -1155,4 +1155,112 @@ private FlowableTransformers() {
         ObjectHelper.verifyPositive(bufferSize, "bufferSize");
         return new FlowableCoalesce<T, R>(null, containerSupplier, coalescer, bufferSize);
     }
+
+    /**
+     * Emits elements into a UnicastProcessor while the given predicate returns true. If the
+     * predicate returns false, a new empty UnicastProcessor is emitted.
+     * @param <T> the source value type
+     * @param predicate the predicate receiving the current value and if returns false,
+     *                  a new buffer is created with the specified item
+     * @return the new FlowableTransformer instance
+     *
+     * @since 0.17.4
+     */
+    @SchedulerSupport(SchedulerSupport.NONE)
+    @BackpressureSupport(BackpressureKind.FULL)
+    public static <T> FlowableTransformer<T, Flowable<T>> windowWhile(Predicate<? super T> predicate) {
+        return windowWhile(predicate, Flowable.bufferSize());
+    }
+
+    /**
+     * Emits elements into a UnicastProcessor while the given predicate returns true. If the
+     * predicate returns false, a new empty UnicastProcessor is emitted.
+     * @param <T> the source value type
+     * @param predicate the predicate receiving the current value and if returns false,
+     *                  a new collection is created with the specified item
+     * @param bufferSize the buffer size hint (the chunk size of the underlying unbounded buffer)
+     * @return the new FlowableTransformer instance
+     *
+     * @since 0.17.4
+     */
+    @SchedulerSupport(SchedulerSupport.NONE)
+    @BackpressureSupport(BackpressureKind.FULL)
+    public static <T> FlowableTransformer<T, Flowable<T>> windowWhile(final Predicate<? super T> predicate, int bufferSize) {
+        Predicate<T> negated = new Predicate<T>() {
```

Comment:
Why is this extra negation present?

Reply:
Because interpreting the predicate differently based on the split Mode (before, after, split) conflates 2 orthogonal concepts and is difficult to understand in the implementation.
Alternatively one could use other names for the Mode, e.g. WHILE, UNTIL, SPLIT_UNTIL or so.

Reply:
This is not a problem in [FlowableBufferPredicate](https://github.com/akarnokd/RxJava2Extensions/blob/master/src/main/java/hu/akarnokd/rxjava2/operators/FlowableBufferPredicate.java#L139) as each case has its own set of operations, "negating" without any extra allocation.

Reply:
Fine, I'll move the negation into the implementation.